great_expectations/datasource/data_connector/inferred_asset_sql_data_connector.py

```diff
@@ -264,6 +264,30 @@ def _introspect_db(
                             }
                         )
 
+        # SQLAlchemy's introspection does not list "external tables" in Redshift Spectrum (tables whose data is stored on S3).
+        # The following code fetches the names of external schemas and tables from a special table
+        # 'svv_external_tables'.
+        try:
+            if "redshift" == engine.dialect.name.lower():
+                result = engine.execute(
+                    "select schemaname, tablename from svv_external_tables"
+                ).fetchall()
+                for row in result:
+                    tables.append(
+                        {
+                            "schema_name": row[0],
+                            "table_name": row[1],
+                            "type": "table",
+                        }
+                    )
+
+        except Exception as e:
+            # If we happen to query a "regular" Redshift (not Spectrum), 'svv_external_tables' table
+            # might not exist. Relying on a specific exception will not work consistently in all psycopg2
+            # versions - falling back on string matching
+            if not "UndefinedTable" in str(e):
```

Comment:
It'd be preferable to do two queries: one checking for table, then another querying the rows. It would be preferable than searching a string for an error